Node/remote-config-server-with-vertex/functions/index.js

```diff
@@ -0,0 +1,119 @@
+// [START remote_config_server_vertex_init]
+const { onRequest } = require("firebase-functions/v2/https");
+const logger = require("firebase-functions/logger");
+
+const { initializeApp } = require("firebase-admin/app");
+const { VertexAI } = require('@google-cloud/vertexai');
+const { getRemoteConfig } = require("firebase-admin/remote-config");
+
+// Set and check environment variables.
+const project = process.env.GCLOUD_PROJECT;
+
+// Initialize Firebase.
+const app = initializeApp();
+// [END remote_config_server_vertex_init]
+
+// [START remote_config_server_vertex_default_values]
+// Define default (fallback) parameter values for Remote Config.
+const defaultConfig = {
+
+  // Default values for Vertex AI.
+  model_name: "gemini-1.5-flash-preview-0514",
+  generation_config: [{
+    "stopSequences": [], "temperature": 0.7,
+    "maxOutputTokens": 64, "topP": 0.1, "topK": 20
+  }],
+  prompt: "I'm a developer who wants to learn about Firebase and you are a \
+    helpful assistant who knows everything there is to know about Firebase!",
+  safety_settings: [{
+    "category":
+      "HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT",
+    "threshold": "HarmBlockThreshold.BLOCK_MEDIUM_AND_ABOVE"
+  }],
+  location: 'us-central1',
+
+  // Disable Vertex AI Gemini API access for testing.
+  vertex_enabled: false
+};
+// [END remote_config_server_vertex_default_values]
+
+// [START remote_config_server_vertex_create_function]
+// Export the function.
+exports.generateWithVertex = onRequest(async (request, response) => {
+
+  try {
```

Comment:
It would be great to add some kind of auth check here to encourage developers to protect their AI endpoints. https://github.com/firebase/functions-samples/pull/1134#discussion_r1583181807 does that with `onCall` + AppCheck + Auth, but this could check Auth with something like: https://firebase.google.com/docs/auth/web/service-worker-sessions#server_side_changes

Reply:
Oh, you do have the `vertex_enabled` flag already. Auth would be ideal, but that flag is probably ok for now

Reply:
But the likelihood of folks leaving vertex_enabled to true exists...so I will get some auth in here! I think the service-worker-sessions method would actually work really well with the tutorial, so I'll try this out.